tests/test_ioutils.py

```diff
@@ -40,6 +41,29 @@ def test_getvalue_rollover(self):
         self.assertEqual(self.spooled_flo.getvalue(), self.test_str)
         self.assertTrue(self.spooled_flo._rolled)
 
+    def test_rollover_custom_directory(self):
+        """dir keyword argument is passed to TemporaryFile instantiation"""
+        custom_dir = mkdtemp()
+        try:
```

Comment:
can be `with tempfile.TemporaryDirectory() as custom_dir:` ... instead of `mkdtemp` + `try...finally` (saves like 3 lines of code!)

https://docs.python.org/3/library/tempfile.html#tempfile.TemporaryDirectory